tests/agent_run_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe "Task 8:", host: :agent do
+  it 'Run the pre-installed agent' do
+    file('/home/learning/.bash_history')
+      .content
+      .should match /sudo puppet agent -t/
+  end
+end
+
+describe "Task 9:", host: :localhost do
+  it 'Sign the agent certificate' do
+    file('/root/.bash_history')
+      .content
+      .should match /puppet cert list/
+    file('/root/.bash_history')
+      .content
+      .should match /puppet cert sign agent.puppet.vm/
```

Comment:
I think there should be an artifact created by signing this certificate. If not, you could use serverspec's exec matcher to run 'puppet cert list' and check that agent.puppet.vm doesn't appear.